google/cloud/aiplatform/models.py

```diff
@@ -2389,3 +2438,624 @@ def export_model(
+                where the user only has a read access.
+            explanation_metadata (explain.ExplanationMetadata):
+                Optional. Metadata describing the Model's input and output for explanation.
+                Both `explanation_metadata` and `explanation_parameters` must be
+                passed together when used. For more details, see
+                `Ref docs <http://tinyurl.com/1igh60kt>`
+            explanation_parameters (explain.ExplanationParameters):
+                Optional. Parameters to configure explaining for Model's predictions.
+                For more details, see `Ref docs <http://tinyurl.com/1an4zake>`
+            project: Optional[str]=None,
+                Project to upload this model to. Overrides project set in
+                aiplatform.init.
+            location: Optional[str]=None,
+                Location to upload this model to. Overrides location set in
+                aiplatform.init.
+            credentials: Optional[auth_credentials.Credentials]=None,
+                Custom credentials to use to upload this model. Overrides credentials
+                set in aiplatform.init.
+            labels (Dict[str, str]):
+                Optional. The labels with user-defined metadata to
+                organize your Models.
+                Label keys and values can be no longer than 64
+                characters (Unicode codepoints), can only
+                contain lowercase letters, numeric characters,
+                underscores and dashes. International characters
+                are allowed.
+                See https://goo.gl/xmQnxf for more information
+                and examples of labels.
+            encryption_spec_key_name (Optional[str]):
+                Optional. The Cloud KMS resource identifier of the customer
+                managed encryption key used to protect the model. Has the
+                form:
+                ``projects/my-project/locations/my-region/keyRings/my-kr/cryptoKeys/my-key``.
+                The key needs to be in the same region as where the compute
+                resource is created.
+
+                If set, this Model and all sub-resources of this Model will be secured by this key.
+
+                Overrides encryption_spec_key_name set in aiplatform.init.
+            staging_bucket (str):
+                Optional. Bucket to stage local model artifacts. Overrides
+                staging_bucket set in aiplatform.init.
+        Returns:
+            model: Instantiated representation of the uploaded model resource.
+        Raises:
+            ValueError: If only `explanation_metadata` or `explanation_parameters`
+                is specified.
+                Also if model directory does not contain a supported model file.
+        """
+        # https://cloud.google.com/vertex-ai/docs/predictions/pre-built-containers#xgboost
+        XGBOOST_SUPPORTED_VERSIONS = ["0.82", "0.90", "1.1", "1.2", "1.3", "1.4"]
```

Comment:
Preference to only maintain one source of truth and reference versions through `aiplatform.constants.prediction._SERVING_CONTAINER_URI_MAP.get('us').get('xgboost').get('cpu').values()`

In this method and other methods.

I acknowledge your concern in this comment: https://github.com/googleapis/python-aiplatform/pull/805/files#r758935431  

For right now since there is no workaround to use this method with containers released after the SDK version is released we can move forward with that current approach.

Reply:
Done.
P.S. The users that install the current version of the SDK will be unable to use the future versions of the frameworks unless they update the SDK.